zenodo/modules/records/serializers/bibtex.py

```diff
@@ -0,0 +1,664 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Zenodo.
+# Copyright (C) 2015, 2016 CERN.
+#
+# Zenodo is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Zenodo is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Zenodo. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Marshmallow based BibTeX serializer for records."""
+
+from __future__ import absolute_import, print_function
+
+import textwrap
+
+from time import strftime
+
+import six
+
+from slugify import slugify
+
+
+class BibTeXSerializer(object):
+    """Marshmallow based BibTeX serializer for records."""
+
+    def serialize(self, pid, record, links_factory=None):
+        """Serialize a single record and persistent identifier.
+
+        :param pid: Persistent identifier instance.
+        :param record: Record instance.
+        :param links_factory: Factory function for record links.
+        """
+        return Bibtex(record=record).format()
+
+    def serialize_search(self, pid_fetcher, search_result, links=None,
+                         item_links_factory=None):
+        """Serialize a search result.
+
+        :param pid_fetcher: Persistent identifier fetcher.
+        :param search_result: Elasticsearch search result.
+        :param links: Dictionary of links to add to response.
+        """
+        records = []
+        for record in search_result['hits']['hits']:
+            records.append(Bibtex(record=record).format())
```

Comment:
I followed the example here: https://github.com/inveniosoftware/invenio-records-rest/blob/master/invenio_records_rest/serializers/dc.py#L64
Maybe it's needed also there the `['source']`?